### PR TITLE
Release 1.87.0

### PR DESCRIPTION
**Pull request by Yiyuanzzz**
### Summary

# 1.87.0
* Feature - Add support for EBS Task Attach on Windows. [#4353](https://github.com/aws/amazon-ecs-agent/pull/4353)
* Enhancement - Validate ECS_OFFHOST_INTROSPECTION_INTERFACE_NAME is a valid interface name. [#4275](https://github.com/aws/amazon-ecs-agent/pull/4275)
* Enhancement - add a warning to ecs-init when it doesnt detect GPU devices and moves on. [#4338](https://github.com/aws/amazon-ecs-agent/pull/4338)
* Enhancement - Add resources to the log message "Resources not consumed, enough resources not available". [#4328](https://github.com/aws/amazon-ecs-agent/pull/4328)
* Bugfix - Add read lock to task object String method. [#4288](https://github.com/aws/amazon-ecs-agent/pull/4288)

### Testing

New tests cover the changes: 

### Additional Information

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
  

**Does this PR include the addition of new environment variables in the README?**

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hozkaya2000 — DISMISSED** [review]

File: agent/version/version.go [hunk 1]
```diff
@@ -22,10 +22,10 @@ package version
...
 // Version is the version of the Agent
-const Version = "1.86.3"
+const Version = "1.87.0"
 
 // GitDirty indicates the cleanliness of the git repo when this agent was built
 const GitDirty = true
 
 // GitShortHash is the short hash of this agent build
-const GitShortHash = "d6e4c7a1"
+const GitShortHash = "f398e88d"
```
Comment (singholt) [inline agent/version/version.go:31]: this should be `c4616021`: commit ID of the latest commit in the [dev branch](https://github.com/aws/amazon-ecs-agent/tree/dev).

your local branch may not be up-to-date with the [last PR](https://github.com/aws/amazon-ecs-agent/pull/4353) that was merged into dev

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,13 @@
 # Changelog
+
+# 1.87.0
+* Feature - Add support for EBS Task Attach on Windows. [#4353](https://github.com/aws/amazon-ecs-agent/pull/4353)
+* Enhancement -  Update go module dependencies. [#4327](https://github.com/aws/amazon-ecs-agent/pull/4327)
```
Comment (singholt) [inline CHANGELOG.md:5]: lets remove this housekeeping PR

**Review by singholt — CHANGES_REQUESTED** [review]

File: agent/version/version.go (hunk 1 above)
Comment (Yiyuanzzz) [reply agent/version/version.go]: Thanks for catching it

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,13 @@
 # Changelog
+
+# 1.87.0
+* Feature - Add support for EBS Task Attach on Windows. [#4353](https://github.com/aws/amazon-ecs-agent/pull/4353)
+* Enhancement -  Update go module dependencies.
```
Comment (singholt) [inline CHANGELOG.md:5]: still lingering around...

Comment (Yiyuanzzz) [reply CHANGELOG.md:5]: oh sorry you mean remove the whole thing

Comment (singholt) [reply CHANGELOG.md]: yes, sorry and thanks!

**Review by singholt — APPROVED** [review]

**Review by mye956 — APPROVED** [review]

**Review by danehlim — APPROVED** [review]

**Pull request merged**